core/iwasm/aot/aot_runtime.c

```diff
@@ -1049,7 +1049,22 @@ memory_instantiate(AOTModuleInstance *module_inst, AOTModuleInstance *parent,
     return NULL;
 }
 
-static AOTMemoryInstance *
+AOTMemoryInstance *
+aot_lookup_memory(AOTModuleInstance *module_inst, char const *name)
+{
+#if WASM_ENABLE_MULTI_MEMORY != 0
+    uint32 i;
+    for (i = 0; i < module_inst->export_memory_count; i++)
+        if (!strcmp(module_inst->export_memories[i].name, name))
+            return module_inst->export_memories[i].memory;
+    return NULL;
+#else
+    (void)module_inst->export_memories;
+    return module_inst->memories[0];
```

Comment:
It's better to check whether `module_inst->memories` is NULL before accessing it, like in `aot_get_default_memory`

The same rationale applies to `wasm_get_default_memory`